pkg/common-controller/snapshot_controller.go

```diff
@@ -1197,21 +1190,23 @@ func (ctrl *csiSnapshotCommonController) SetDefaultSnapshotClass(snapshot *crdv1
 		return nil, snapshot, nil
 	}
 
-	storageclass, err := ctrl.getStorageClassFromVolumeSnapshot(snapshot)
+	// Find default snapshot class if available
+	list, err := ctrl.classLister.List(labels.Everything())
 	if err != nil {
 		return nil, nil, err
 	}
-	// Find default snapshot class if available
-	list, err := ctrl.classLister.List(labels.Everything())
+
+	pvDriver, err := ctrl.pvDriverFromSnapshot(snapshot)
 	if err != nil {
+		klog.Errorf("failed to get pv csi driver from snapshot %s-%s: %q", snapshot.Namespace, snapshot.Name, err)
```

Comment:
nit: It may be worth having consistent formatting of snapshot strings (`%s/%s` above vs. `%s-%s` here).

Reply:
done